cdisc_rules_engine/models/failed_validation_entity.py

```diff
@@ -1,21 +1,27 @@
+from dataclasses import dataclass, field
 from cdisc_rules_engine.enums.execution_status import ExecutionStatus
-
 from .base_validation_entity import BaseValidationEntity
 
 
+@dataclass
 class FailedValidationEntity(BaseValidationEntity):
     """
     The entity describes an error that occurred during validation
     indicating that the process has finished its execution with error.
     """
 
+    dataset: str = field(init=False)
+    _error: str = field(init=False)
+    _message: str = field(init=False)
+    
     def __init__(self, error: str, message: str, dataset: str):
```

Comment:
Where possible, do not explicitly define an __init__ method. you should be able to do this work when you assign the fields. The calls to the constructors will need to be changed and this is okay.